src/main/java/org/folio/service/processing/ParallelFileChunkingProcessor.java

```diff
@@ -66,23 +70,27 @@ public ParallelFileChunkingProcessor(Vertx vertx, KafkaConfig kafkaConfig) {
 
   @Override
   public void process(JsonObject jsonRequest, JsonObject jsonParams) { //NOSONAR
-    ProcessFilesRqDto request = jsonRequest.mapTo(ProcessFilesRqDto.class);
-    UploadDefinition uploadDefinition = request.getUploadDefinition();
-    JobProfileInfo jobProfile = request.getJobProfileInfo();
-    OkapiConnectionParams params = new OkapiConnectionParams(jsonParams.mapTo(HashMap.class), this.vertx);
-    UploadDefinitionService uploadDefinitionService = new UploadDefinitionServiceImpl(vertx);
-    succeededFuture()
-      .compose(ar -> uploadDefinitionService.getJobExecutions(uploadDefinition, params))
-      .compose(jobExecutions -> updateJobsProfile(jobExecutions, jobProfile, params))
-      .compose(ar -> FileStorageServiceBuilder.build(this.vertx, params.getTenantId(), params))
-      .compose(fileStorageService -> {
-        processFiles(jobProfile, uploadDefinitionService, fileStorageService, uploadDefinition, params);
-        uploadDefinitionService.updateBlocking(
-          uploadDefinition.getId(),
-          definition -> succeededFuture(definition.withStatus(UploadDefinition.Status.COMPLETED)),
-          params.getTenantId());
-        return succeededFuture();
-      });
+    try {
+      ProcessFilesRqDto request = jsonRequest.mapTo(ProcessFilesRqDto.class);
+      UploadDefinition uploadDefinition = request.getUploadDefinition();
+      JobProfileInfo jobProfile = request.getJobProfileInfo();
+      OkapiConnectionParams params = new OkapiConnectionParams(jsonParams.mapTo(HashMap.class), this.vertx);
+      UploadDefinitionService uploadDefinitionService = new UploadDefinitionServiceImpl(vertx);
+      succeededFuture()
+        .compose(ar -> uploadDefinitionService.getJobExecutions(uploadDefinition, params))
+        .compose(jobExecutions -> updateJobsProfile(jobExecutions, jobProfile, params))
+        .compose(ar -> FileStorageServiceBuilder.build(this.vertx, params.getTenantId(), params))
+        .compose(fileStorageService -> {
+          processFiles(jobProfile, uploadDefinitionService, fileStorageService, uploadDefinition, params);
+          uploadDefinitionService.updateBlocking(
+            uploadDefinition.getId(),
+            definition -> succeededFuture(definition.withStatus(UploadDefinition.Status.COMPLETED)),
+            params.getTenantId());
+          return succeededFuture();
+        });
+    } catch (Exception e) {
+      LOGGER.error("Can`t process file. Cause: ", e.getMessage());
```

Comment:
do we have JobExecution entity created at this point? if yes - we need to update its status to failed

Reply:
@KaterynaSenchenko  it seems like we don`t have this property here. I changed this try/catch to the ".onFailure()" with logging.

 I want to mention, that if there are some exceptions in the places where we have JobExecution entity - it`ll log and update status at the level below (processFile-method) because I added there one more error catching in the previous tickets (which fixed some bugs when data-import can`t started).